deepchem/dock/docking.py

```diff
@@ -105,11 +118,15 @@ def dock(self,
       complexes, scores = outputs
     else:
       complexes = outputs
+
     # We know use_pose_generator_scores == False in this case
     if self.scoring_model is not None:
       for posed_complex in complexes:
+        # NOTE: this casting is workaround. This line doesn't effect anything to the runtime
+        self.featurizer = cast(ComplexFeaturizer, self.featurizer)
         # TODO: How to handle the failure here?
-        features, _ = self.featurizer.featurize([molecular_complex])
+        (protein_file, ligand_file) = molecular_complex
+        features, _ = self.featurizer.featurize([protein_file], [ligand_file])
```

Comment:
The original code may be bug... Is it correct...?

Reply:
mypy taught me this bug!  Introducing type annotation leads to solve the potential bug.